packages/traceability/issuer-irec-api-wrapper/src/IRECAPIClient.ts

```diff
@@ -457,6 +464,7 @@ export class IRECAPIClient {
             response.data.refresh_token,
             response.data.expires_in
         );
+        this.emit('tokensRefreshed', this.accessTokens);
```

Comment:
Not sure why we need to use the mechanism of event emitting here? Could you explain your thought process here?

Reply:
Since we are storing all the access tokens we have to update them in DB every time it refreshes. So events are the obvious approach to use.
If you have a better idea, I'm ready to listen.